contracts/L2Deposit.sol

```diff
@@ -0,0 +1,201 @@
+import "./interfaces/ISpender.sol";
+import "./utils/BaseLibEIP712.sol";
+import "./utils/Ownable.sol";
+import "./utils/L2DepositLibEIP712.sol";
+import "./utils/SignatureValidator.sol";
+
+contract L2Deposit is IL2Deposit, ReentrancyGuard, Ownable, BaseLibEIP712, SignatureValidator {
+    using SafeERC20 for IERC20;
+
+    // Peripherals
+    address public immutable userProxy;
+    IPermanentStorage public immutable permStorage;
+
+    // Bridges
+    IArbitrumL1GatewayRouter public immutable arbitrumL1GatewayRouter;
+    IArbitrumL1Inbox public immutable arbitrumL1Inbox;
+    IOptimismL1StandardBridge public immutable optimismL1StandardBridge;
+
+    // Below are the variables which consume storage slots.
+    ISpender public spender;
+
+    // Refund collector contract address on Arbitrum
+    address public arbitrumL2RefundCollector;
+
+    constructor(
+        address _owner,
+        address _userProxy,
+        ISpender _spender,
+        IPermanentStorage _permStorage,
+        IArbitrumL1GatewayRouter _arbitrumL1GatewayRouter,
+        IArbitrumL1Inbox _arbitrumL1Inbox,
+        address _arbitrumL2RefundCollector,
+        IOptimismL1StandardBridge _optimismL1StandardBridge
+    ) Ownable(_owner) {
+        userProxy = _userProxy;
+        spender = _spender;
+        permStorage = _permStorage;
+        arbitrumL1GatewayRouter = _arbitrumL1GatewayRouter;
+        arbitrumL1Inbox = _arbitrumL1Inbox;
+        arbitrumL2RefundCollector = _arbitrumL2RefundCollector;
+        optimismL1StandardBridge = _optimismL1StandardBridge;
+    }
+
+    modifier onlyUserProxy() {
+        require(address(userProxy) == msg.sender, "L2Deposit: not the UserProxy contract");
+        _;
+    }
+
+    function setArbitrumL2RefundCollector(address _newArbitrumL2RefundCollector) external onlyOwner {
+        require(_newArbitrumL2RefundCollector != address(0), "L2Deposit: Arbitrum L2 refund collector cannot be zero address");
+        arbitrumL2RefundCollector = _newArbitrumL2RefundCollector;
+
+        emit SetArbitrumL2RefundCollector(_newArbitrumL2RefundCollector);
+    }
+
+    function upgradeSpender(address _newSpender) external onlyOwner {
+        require(_newSpender != address(0), "L2Deposit: spender can not be zero address");
+        spender = ISpender(_newSpender);
+
+        emit UpgradeSpender(_newSpender);
+    }
+
+    function collectArbitrumL2Refund(
+        uint256 amount,
+        uint256 maxSubmissionCost,
+        uint256 maxGas,
+        uint256 gasPriceBid
+    ) external onlyOwner {
+        uint256 seqNum = arbitrumL1Inbox.createRetryableTicket(
+            arbitrumL2RefundCollector,
+            amount,
+            maxSubmissionCost,
+            arbitrumL2RefundCollector,
+            arbitrumL2RefundCollector,
+            maxGas,
+            gasPriceBid,
+            bytes("")
+        );
+
+        emit CollectArbitrumL2Refund(arbitrumL2RefundCollector, seqNum, amount, maxSubmissionCost, maxGas, gasPriceBid);
+    }
```

Comment:
In Arbitrum Nitro inbox would collect submission cost on L1. This function might need to be payable to provide call value for `createRetryableTicket` function call.

This might need to confirm on testnet. (Rinkeby or Goerli)

Reply:
Checked and yes `createRetryableTicket` should be called with non-zero `value` otherwise will be reverted. Fixed and add tests in 68caea0.